src/config.py

```diff
@@ -11,6 +11,9 @@ class Config:
 
     MONGOS_PORT = 27018
     MONGOS_SOCKET = "/var/snap/charmed-mongodb/common/var/mongodb-27018.sock"
+    MONGOS_SOCKET_URI_FMT = (
+        "%2Fvar%2Fsnap%2Fcharmed-mongodb%2Fcommon%2Fvar%2Fmongodb-27018.sock"
+    )
```

Comment:
you can simplify this with the following:
```
from urllib.parse import quote

MONGOS_SOCKET_URI_FMT = quote(MONGOS_SOCKET, safe='')
```

Reply:
TIL!